tesseract_kinematics/core/src/kinematic_group.cpp

```diff
@@ -34,8 +34,67 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
 #include <tesseract_common/utils.h>
 
 #include <tesseract_scene_graph/graph.h>
+#include <tesseract_scene_graph/joint.h>
 #include <tesseract_state_solver/state_solver.h>
 
+/**
+ * @brief Returns the names of the links that comprise a fixed-joint kinematic tree with the input link
+ * @details This method first traverses from the input link "up" the scene graph hierarchy, following the link's
+ * in-bound joints, to find the input link's highest level parents that are connected to it by fixed joints. Next, it
+ * identifies all the fixed-joint children of these highest level parent links. Together, the highest level parent links
+ * and their fixed-joint children comprise a set of links (including the input link), between all of which exists a tree
+ * of fixed joints. If the link does not have any fixed-joint parents, the name of the link itself will be returned.
```

Comment:
Why not just return an empty vector?

Reply:
Mostly for convenience. In the class constructor I push the contents of the returned vector into `working_frames_` and `inv_kin_tip_links_map_`, so it's more convenient to also have the nominal working frame/tip link be a part of that vector than having to add it to those class members separately. I tried to rename this method and update its documentation to indicate that the return value is the set of links involved in a fixed joint kinematic tree that includes the nominal link itself; I can add more detail there if you want